script.js

```diff
@@ -1,4 +1,142 @@
+      return `Player 2: Your number is ${player2CombinedNumber}.Now let's see who wins! Please click submit to see the result`;
+    }
+  }
+};
+
+const toCompareNumber = (player1number, player2number) => {
+  return player1number == player2number
+    ? "It is a draw"
+    : player1number > player2number
+    ? "Player 1 wins"
+    : "Player 2 wins";
+};
+
+//to change game mode: default is waiting for user to key in name
+const changeGameMode = (userInput) => {
+  //input for user to key in name
+  if (GAME_STATE == GAMEMODE_WAITING_FOR_NAME1) {
+    if (Number.isNaN(Number(userInput)) == false) {
+      return "This is not a valid input, please key in your name.";
+    } else {
+      var player1Name = userInput;
+      //to choose game player and to choose game state; now Player 1 mode and to choose player
+      GAMEPLAYER = GAMEMODE_PLAYER1;
+      GAME_STATE = GAMEMODE_T0_CHOOSE_PLAYER;
+      return `PLAYER 1<br> ------------------------ <br> Hi ${player1Name}. You are first! <br> Please click submit to roll the dice.`;
+    }
+  }
+
+  // now game state is to choose player
+
+  if (GAME_STATE == GAMEMODE_T0_CHOOSE_PLAYER) {
+    if (GAMEPLAYER == GAMEMODE_PLAYER1 || GAMEPLAYER == GAMEMODE_PLAYER2) {
+      if (userInput == "") {
+        var playerdiceRollresult = showsDiceRollsResult();
+        // To output the results of the dice roll and for user to input 1/2 for order of dice roll
+        GAME_STATE = GAMEMODE_T0_CHOOSE_ORDER;
+        return (
+          playerdiceRollresult +
+          `<br> <br>Please key in <1> or <2> to choose the order of your dice: <br> <1>: For number corresponding on dice 1 to go first.<br> <2>: For number corresponding on dice 2 to go first.<br><br> Please note the higher combined number will win.`
+        );
+      }
+    }
+  }
+
+  // now game state is to choose order
+  if (GAME_STATE == GAMEMODE_T0_CHOOSE_ORDER) {
+    return showGamePlayerChoice(GAMEPLAYER, userInput);
+  }
+
+  //game state for player 2 to key in name
+  if (GAME_STATE == GAMEMODE_WAITING_FOR_NAME2) {
+    var player2Name = userInput;
+    GAME_STATE = GAMEMODE_T0_CHOOSE_PLAYER;
+    GAMEPLAYER = GAMEMODE_PLAYER2;
+    return `PLAYER 2<br> ------------------------ <br> Hi ${player2Name}. You are next <br> Please click submit to roll the dice.`;
+  }
+
+  if (GAME_STATE == TO_CHOOSE_WINNER) {
+    GAME_STATE = GAMEMODE_T0_CHOOSE_PLAYER;
+    GAMEPLAYER = GAMEMODE_PLAYER1;
+    if (userInput == "") {
+      return `${toCompareNumber(
+        player1CombinedNumber,
+        player2CombinedNumber
+      )} <br> Player 1's number is ${player1CombinedNumber}.<br>
+      Player 2's number is ${player2CombinedNumber}.<br><br> Please click submit for player 1 to roll the dice again `;
+    }
+  }
+  player2CombinedNumber = 0;
+  player1CombinedNumber = 0;
+  //to restart the game again back to player 1
+};
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var result = changeGameMode(input);
```

Comment:
i would consider naming the function changeGameMode to gameStart instead. Might be clearer.